android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MainActivity.java

```diff
@@ -1092,28 +1110,39 @@ public void onClick(DialogInterface dialog, int which) {
                 } else {
                     oldLatLng = new LatLng(meHelper.mLocationInSimulator.getLatitude(), meHelper.mLocationInSimulator.getLongitude());
                 }
-                switch (which) {
-                    case 0:
-                        Log.i(TAG, "Spoofing GPS at "+location);
-                        showMessage("GPS spoofing activated.");
-                        float[] results = new float[1];
-                        Location.distanceBetween(oldLatLng.latitude, oldLatLng.longitude, spoofLatLng.latitude, spoofLatLng.longitude, results);
-                        double distance = results[0]/1000;
-                        initUserMobileIcon();
-                        mUserLocationMarker.setSnippet("Spoofed "+String.format("%.2f", distance)+" km from actual location");
-                        meHelper.setSpoofedLocation(location);
-                        resetPosition[0] = false;
-                        break;
-                    case 1:
-                        Log.i(TAG, "Update GPS in simulator to "+location);
-                        initUserMobileIcon();
-                        mUserLocationMarker.setSnippet((String) getResources().getText(R.string.drag_to_spoof));
-                        updateLocSimLocation(mUserLocationMarker.getPosition().latitude, mUserLocationMarker.getPosition().longitude);
-                        meHelper.setSpoofedLocation(location);
-                        resetPosition[0] = false;
-                        break;
-                    default:
-                        Log.i(TAG, "Unknown dialog selection.");
+
+                String selectedItemText = items.get(which);
+
+                if (selectedItemText.equals(spoofText)) {
+                    Log.i(TAG, "Spoofing GPS at " + location);
+                    showMessage("GPS spoofing activated.");
+                    float[] results = new float[1];
+                    Location.distanceBetween(oldLatLng.latitude, oldLatLng.longitude, spoofLatLng.latitude, spoofLatLng.longitude, results);

```

Comment:
is this a built in function? Seems odd to return the distance between 2 points in an array

Reply:
Yes, it's built in. https://developer.android.com/reference/android/location/Location.html#distanceBetween(double,%20double,%20double,%20double,%20float[])

Reply:
The other array entries are for bearings.